packages/protocol/contracts/governance/Election.sol

```diff
@@ -261,10 +274,23 @@ contract Election is
       alreadyVotedForGroup = alreadyVotedForGroup || groups[i] == group;
     }
     if (!alreadyVotedForGroup) {
-      require(groups.length < maxNumGroupsVotedFor, "Voted for too many groups");
+      require(
+        allowedToVoteOverMaxNumberOfGroups[msg.sender] || groups.length < maxNumGroupsVotedFor,
+        "Voted for too many groups"
+      );
       groups.push(group);
     }
 
+    if (groups.length > maxNumGroupsVotedFor) {
+      CachedVotes storage cachedVotes = cachedVotesByAccount[account];
+      if (cachedVotesByAccount[account].totalVotes > 0) {
+        // if total account votes were counted at least once,
+        // lets add new votes to cache for possible slashing
```

Comment:
What does slashing have to do with this?

Reply:
Same question. Should we just update the cache (using `updateTotalVotesByAccountForGroup`) here when `allowedToVoteOverMaxNumberOfGroups` is true (whether or not we are voting over the limit)?

Reply:
In fact, could/should we just update the cache in the low-level increment/decrement total votes functions?